cmd/web-handlers.go

```diff
@@ -1339,19 +1341,22 @@ func (web *webAPIHandlers) DownloadZip(w http.ResponseWriter, r *http.Request) {
 			}
 			var writer io.Writer
 
-			if info.IsCompressed() {
+			if comp, err := info.IsCompressedOK(); comp {
+				if err != nil {
+					return err
+				}
 				// Open a pipe for compression
 				// Where compressWriter is actually passed to the getObject
 				decompressReader, compressWriter := io.Pipe()
-				snappyReader := snappy.NewReader(decompressReader)
+				s2Reader := s2.NewReader(decompressReader)
```

Comment:
I think this is doing double decompression - or it is missing some decryption code... something is wrong here (in the existing code as well)